### PR TITLE
Make various test CA certs RFC 5280 compliant w.r.t. X.509 extensions

### PR DESCRIPTION
**Pull request by DDvO**
This is a spin-off of #13658 to reduce the size of that PR.

- [x] tests are added or updated

### CODE REVIEW

File: test/certs/setup.sh [hunk 1]
```diff
@@ -410,7 +410,7 @@ openssl req -new -noenc -subj "/CN=localhost" \
 ./mkcert.sh genct server.example embeddedSCTs1-key embeddedSCTs1 embeddedSCTs1_issuer-key embeddedSCTs1_issuer ct-server-key
 
 OPENSSL_SIGALG=ED448 OPENSSL_KEYALG=ed448 ./mkcert.sh genroot "Root Ed448" \
-    root-ed448-key root-ed448-cert
+    root-ed448-key root-ed448-cert # TODO updating this cert does not work
```
Comment (t8m) [inline test/certs/setup.sh:413]: Better than adding TODO to the code would be to create and issue.

Comment (t8m) [reply test/certs/setup.sh:413]: Hmm, actually it is just sufficient to replace `OPENSSL_SIGALG=ED448` with `OPENSSL_SIGALG=`

Comment (DDvO) [reply test/certs/setup.sh]: Yep, would be nice if we can solve this right away.

With the trick you provided I now get a new cert (pushed as part of the fixup PR), but
```
apps/openssl x509 -noout -text -in test/certs/root-ed448-cert.pem
```
yields
```
Could not read certificate from test/certs/root-ed448-cert.pem
Unable to load certificate

Comment (t8m) [reply test/certs/setup.sh]: Weird, I do not see this issue with the current master. Also you seem to have produced empty file test/certs/root-ed448-cert.pem in the fixup.

Comment (DDvO) [reply test/certs/setup.sh]: Argh, my `mkcert.sh` run had picked up an old openssl app version.
Now it works for me, too; pushed the resulting cert.

**Review by t8m — APPROVED** [review]

**Comment by DDvO** [comment]
Merged - thanks @t8m

**Pull request closed without merging**